### DISCUSSION TITLE
Component system.

### DISCUSSION THREAD

**Discussion by ltdrdata** [Comfy-Org/ComfyUI#294] [Ideas]
I think one workflow can be act as a component.

1. Add collapse feature to Group.
2. Make workflow on one group and collaspe it with name. And save it.
3. Add incremental load feature to main menu. (don't clear current workflow.)
4. Now we can expand our workflow easily.

We have several choices for representing of collapsing group.

1. expose unplugged inputs/outputs on collapsed group
2. expose uni-links for inputs and outputs. And provide linking capabilty on only uncollapsed mode.

**Comment by WASasquatch** [comment]
This is a great idea, especially for simply packaging up "custom node networks" that do specific tasks for people.

I do wonder if it's possible in the current setup.

I think that we shouldn't be using JSON but XML for workflow files, as Python, and everything else has powerful DOM manipulators to merge, expand, add to parents, etc that's designed for this sort of data storage. In fact some commercial products use XML for their node networks too. 

With XML, a node could be a single tag, without the need for various dictionaries for one thing. It's also much more human readable for editing a XML file if a workflow is bugged for some reason and not loading properly. Many IDE editors recognize unclosed tags, and other common user mistakes much better with markdown XML then arrays too.

**Comment by ltdrdata** [comment]
We can go step-by-step.

For now, it is sufficient to only provide the functionality of accumulating a saved workflow into an empty area of the current workflow without erasing it.

In our next step, we can focus on improving the data structure to better accommodate our future requirements.
